### PR TITLE
mdt/mtc_config: Fix testcase failure issues

### PR DESCRIPTION
**Pull request by glb1013**
check input argement, if NULL, return EINVAL; in test case, change the buf assiged to configdata to be writable array;
Signed-off-by: glb1013@163.com

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [73039922f8ce1c49370a82b3156984d98e477af2] - Code Rule Check Result:
os/fs/driver/mtd/mtd_config.c:1170: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/fs/driver/mtd/mtd_config.c:1182: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [73039922f8ce1c49370a82b3156984d98e477af2] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 1]
```diff
@@ -1635,7 +1635,7 @@ static void tc_driver_mtd_config_ops(void)
 	int fd;
 	int ret;
 	struct config_data_s config;
-	char *buf = "test";
+	char buf[8] = "test";
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Unnecessary change, I think

Comment (Taejun-Kwon) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: Please revert file access type (10644 is correct)

File: os/fs/driver/mtd/mtd_config.c [hunk 2]
```diff
@@ -1167,15 +1167,23 @@ static int mtdconfig_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 		/* Set the config item */
 
 		pdata = (FAR struct config_data_s *)arg;
-		ret = mtdconfig_setconfig(dev, pdata);
+		if(pdata != NULL) {
```
Comment (Taejun-Kwon) [inline os/fs/driver/mtd/mtd_config.c]: please see below
```
if (arg == 0) {
   fdbg("Error blah blah blah...\n");
   return -EINVAL;
}

pdata = (FAR struct config_data_s *)arg;
....

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 1 above)
Comment (glb1013) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: In another mcu, the string is allocated in the flash memory, and the pointer "char *buf" points to the readonly memeory. So in the test case "ret = ioctl(fd, CFGDIOC_GETCONFIG, &config);", an exception is popped up in memcpy() invoked by mtdconfig_readbytes() after the entry is found, because it tries to write data to readonly memory. 
It is better to define the buf as array allocated in stack to make sure the memory is writable and it will be safe for all mcus.

**Comment by seinfra** [comment]
Target : [9f2df9eab10ed8546809bacb42910156c7234dcf] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9f2df9eab10ed8546809bacb42910156c7234dcf] - Code Rule Check (C++) OK.

File: os/fs/driver/mtd/mtd_config.c (hunk 2 above)
Comment (glb1013) [reply os/fs/driver/mtd/mtd_config.c]: Fixed

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 1 above)
Comment (glb1013) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: File access type is fixed.

File: os/fs/driver/mtd/mtd_config.c [hunk 3]
```diff
@@ -1165,6 +1165,10 @@ static int mtdconfig_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 	case CFGDIOC_SETCONFIG:
 
 		/* Set the config item */
+		if (arg == 0) {
```
Comment (Taejun-Kwon) [inline os/fs/driver/mtd/mtd_config.c]: I mean check this before switch.

Comment (glb1013) [reply os/fs/driver/mtd/mtd_config.c]: Done.

**Comment by seinfra** [comment]
Target : [e9b52489b99de0a110fc5e44165d343a1b17395a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e9b52489b99de0a110fc5e44165d343a1b17395a] - Code Rule Check OK.

**Comment by Taejun-Kwon** [comment]
mtd/mtd_config is correct.

**Comment by Taejun-Kwon** [comment]
This is Old PR, but do not apply main comment.
Hence close this. please re-open when you ready.
